qt/scientific_interfaces/ISISReflectometry/GUI/Instrument/InstrumentPresenter.cpp

```diff
@@ -23,16 +24,33 @@ InstrumentPresenter::InstrumentPresenter(IInstrumentView *view,
                                          Instrument instrument)
     : m_view(view), m_model(std::move(instrument)) {
   m_view->subscribe(this);
+}
+
+void InstrumentPresenter::acceptMainPresenter(
```

Comment:
There are a few small chunks of repeated code across the presenters. Would it make sense to move these to a bare-bones class for the various presenters to inherit from? (This may well break some best practice I'm not aware of)